src/CardDav/Backend.php

```diff
@@ -244,10 +243,10 @@ private function getLinkedData(string $uri)
      * ->LASTNAME, ->FIRSTNAME etc. extracted from ->N
      * ->PHOTO with embedded data from linked sources (equal for KEY, LOGO or SOUND)
      *
-     * @param Document $vcard
-     * @return Document
+     * @param mixed $vcard
+     * @return mixed
      */
-    public function enrichVcard(Document $vcard): Document
```

Comment:
Warum müssen die type hints geändert werden?

Reply:
Gemäß der Anleitung [http://sabre.io/vobject/vcard/](http://sabre.io/vobject/vcard/) habe ich den use geändert nur auf
http://sabre.io/vobject/vcard/
Darum funktioniert der type hint auf vobject/vcard/Document nicht (mehr).
Ich denke es macht keinen großen Unterschied für andere Nutzer ob da nun mixed oder Document steht.

Reply:
Ich sehe da keine Anleitung die das erforderlich macht. Solange `VObject\Document` in die Funktion rein geht und nichts anderes erwartet wird sollte der Typ auch so bleiben. Oder gibts damit irgendein Problem?

> Darum funktioniert der type hint auf vobject/vcard/Document nicht (mehr).

Du hast das use entfernt, deshalb ist der Typ nicht bekannt, das ist wenig verwirrend.

Welches Problem willst Du hier wirklich lösen- der komplette Code hat doch funktioniert?